.github/workflows/tests.yml

```diff
@@ -0,0 +1,40 @@
+name: tests
+
+on:
+  push: null
+
+jobs:
+  tests:
+    name: tests
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: goanpeca/setup-miniconda@v1
+        with:
+          python-version: 3.7
+          channels: conda-forge,defaults
+          channel-priority: strict
+          show-channel-urls: true
+
+      - name: install code and deps
+        shell: bash -l {0}
+        run: |
+          conda install --file requirements.txt black coverage coveralls mock pytest pytest-cov pip ${CONDA_PKGS} --yes --quiet
+          python -m pip install -e .
+          git config --global user.email "smithy@smithy.smithy"
+          git config --global user.name "smithy"
+
+      - name: run tests
+        shell: bash -l {0}
+        run: |
+          py.test tests --cov conda_smithy
+          black --check --verbose --diff --config=pyproject.toml .
```

Comment:
Out of the scope of this PR but should we add pre-commits here? It may be a bit late for that and it could disrupt more than it could help.

Reply:
Yeah, I'd say we punt on that. We might need a big diff to get it to pass.